File: Documentation Code Snippets/ScanbotSDK Examples/Barcode Scanner/Barcode and QR-Code Scanner/DetectionOnImage/Objc/BarcodeDetectionOnImage.m
//
//  BarcodeDetectionOnImage.m
//  ScanbotSDK Examples
//
//  Created by Rana Sohaib on 24.03.23.
//

#import "BarcodeDetectionOnImage.h"
@import ScanbotSDK;

@implementation BarcodeDetectionOnImage

- (void) detectBarcodesOnImage {
    
    // Image containing barcode.
    UIImage *image = [UIImage imageNamed:@"barcodeImage"];
    
    // Types of barcodes you want to detect.
    NSArray<SBSDKBarcodeType *> *typesToDetect = [SBSDKBarcodeType allTypes];
    
    // Creates an instance of `SBSDKBarcodeScanner`.
    SBSDKBarcodeScanner *detector = [[SBSDKBarcodeScanner alloc] initWithTypes:typesToDetect];
    
    // Returns the result after running detector on the image.
    NSArray<SBSDKBarcodeScannerResult *> *results = [detector detectBarCodesOnImage:image];
}

@end

File: Documentation Code Snippets/ScanbotSDK Examples/Barcode Scanner/Barcode and QR-Code Scanner/DetectionOnImage/Swift/BarcodeDetectionOnImage.swift
//
//  BarcodeDetectionOnImage.swift
//  ScanbotSDK Examples
//
//  Created by Rana Sohaib on 24.03.23.
//

import Foundation
import ScanbotSDK

func detectBarcodesOnImage() {
    
    // Image containing barcode.
    guard let image = UIImage(named: "barcodeImage") else { return }
    
    // Types of barcodes you want to detect.
    let typesToDetect = SBSDKBarcodeType.allTypes
    
    // Creates an instance of `SBSDKBarcodeScanner`.
    let detector = SBSDKBarcodeScanner(types: typesToDetect)
    
    // Returns the result after running detector on the image.
    let result = detector.detectBarCodes(on: image)
}
